File: app/src/main/java/me/gabeg/sicksends/SsSportViewModel.java
package me.gabeg.sicksends;

import android.app.Application;

/**
 * Sport view model.
 */
public class SsSportViewModel
	extends SsProblemViewModel<SsSportRepository, SsSport>
{

	/**
	 */
	public SsSportViewModel(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemViewModel#buildRepository(Application)
	 */
	public void buildRepository(Application app)
	{
		this.mRepository = new SsSportRepository(app);
	}

}

File: app/src/main/java/me/gabeg/sicksends/db/sport/SsSportViewModel.kt
package me.gabeg.sicksends.db.sport

import androidx.lifecycle.SavedStateHandle
import dagger.hilt.android.lifecycle.HiltViewModel
import me.gabeg.sicksends.db.generic.SsGenericProblemViewModel
import javax.inject.Inject

/**
 * Sport view model.
 */
@HiltViewModel
class SsSportViewModel @Inject constructor(
	savedStateHandle: SavedStateHandle,
	repo : SsSportRepository)
		: SsGenericProblemViewModel<SsSportProblem>(savedStateHandle, repo)
